src/node/rpc/jwt_management.h

```diff
@@ -21,18 +22,62 @@
 
 namespace ccf
 {
+  static bool check_issuer(
+    const std::string& issuer, const std::string& constraint)
+  {
+    const auto issuer_domain = http::parse_url_full(issuer).host;
+    const auto constraint_domain = http::parse_url_full(constraint).host;
+
+    if (constraint_domain.empty())
+    {
+      return false;
+    }
+
+    // Either constraint's domain == issuer's domain or it is a subdomain, e.g.:
+    // limited.facebok.com
+    //        .facebok.com
+    if (issuer_domain != constraint_domain)
+    {
+      const auto start_seek =
+        issuer_domain.size() - (constraint_domain.size() + 1);
+      const auto count_seek = constraint_domain.size() + 1;
+      const auto pattern = "." + constraint_domain;
+
+      return start_seek > 0 // at least one letter preceeds .issuer.domain
+        && issuer_domain.substr(start_seek, count_seek) == pattern;
```

Comment:
Could this be replaced with `ends_with()`?

```suggestion
      return issuer_domain.ends_with("." + constraint_domain);
```

Reply:
Indeed, I missed we got it in our current std version.